fapolicy_analyzer/ui/searchable_list.py

```diff
@@ -152,6 +152,12 @@ def on_view_selection_changed(self, selection):
         data = [model[i] for i in treeiter] if model and treeiter else []
         self.selection_changed(data)
 
-    def on_search_changed(self, search):
+    def on_search_icon_press(self, *args):
+        self.get_object("search").set_text("")
         self.treeViewFilter.refilter()
         self._update_tree_count(self.__get_tree_count())
```

Comment:
To DRY this code up a bit, make a private method called `__refilter` or something and have both handlers call it.